CorsixTH/Lua/dialogs/menu.lua

```diff
@@ -545,11 +545,94 @@ end
 function UIMenuBar:makeMapeditorMenu(app)
   local menu = UIMenu()
   local hotkeys = app.hotkeys
-
   menu:appendItem(_S.menu_file.load:format(hotkey_value_label("ingame_loadMenu", hotkeys)), function() self.ui:addWindow(UILoadMap(self.ui, "map")) end)
     :appendItem(_S.menu_file.save:format(hotkey_value_label("ingame_saveMenu", hotkeys)), function() self.ui:addWindow(UISaveMap(self.ui)) end)
+	:appendItem(_S.menu_file.restart:format(hotkey_value_label("ingame_restartLevel", hotkeys)), function() app:mapEdit() end)
     :appendItem(_S.menu_file.quit:format(hotkey_value_label("ingame_quitLevel", hotkeys)), function() self.ui:quit() end)
   self:addMenu(_S.menu.file, menu)
+  local options = UIMenu()
+  if app.audio.has_bg_music then
+    local function vol(level, setting)
+      if setting == "music" then
+        return level == app.config.music_volume,
+          function()
+            app.audio:setBackgroundVolume(level)
+          end,
+          ""
+      elseif setting == "sound" then
+        return level == app.config.sound_volume,
+          function()
+            app.audio:setSoundVolume(level)
+          end,
+          ""
+      else
+        return level == app.config.announcement_volume,
+          function()
+            app.audio:setAnnouncementVolume(level)
+          end,
+          ""
+      end
+    end
+
+    local function appendVolume(setting)
+      local volume_menu = UIMenu() -- The three Volume menus
+      for level = 10, 100, 10 do
+        volume_menu:appendCheckItem(_S.menu_options_volume[level], vol(level / 100, setting))
+      end
+      return volume_menu
+    end
+  options:appendCheckItem(_S.menu_options.sound:format(hotkey_value_label("ingame_toggleSounds", hotkeys)),
+    app.config.play_sounds,
+    function(item)
+      app.audio:playSoundEffects(item.checked)
+    end,
+    nil,
+    function()
+      return app.config.play_sounds
+    end)
+
+
+ options:appendCheckItem(_S.menu_options.music:format(hotkey_value_label("ingame_toggleMusic", hotkeys)),
+    app.config.play_music,
+    function(item)
+      app.config.play_music = item.checked
+      self.ui:togglePlayMusic(item)
+    end,
+    nil,
+    function()
+      return app.config.play_music
+    end)
+
+  options
+    :appendMenu(_S.menu_options.sound_vol,         appendVolume("sound"))
+    :appendMenu(_S.menu_options.music_vol,         appendVolume("music"))
+    :appendItem(_S.menu_options.jukebox:format(hotkey_value_label("ingame_jukebox", hotkeys)), function() self.ui:addWindow(UIJukebox(app)) end)
```

Comment:
```suggestion
  options:appendMenu(_S.menu_options.sound_vol, appendVolume("sound"))
    :appendMenu(_S.menu_options.music_vol, appendVolume("music"))
    :appendItem(_S.menu_options.jukebox:format(hotkey_value_label("ingame_jukebox", hotkeys)), function() self.ui:addWindow(UIJukebox(app)) end)
```

Sorry, never been keen on the excessive align for parts of the function params.

Reply:
To note: 608 could be split across two lines but that's a smaller issue and one that can be left for now.